cpacker/MemGPT
# fix: Fix load bug

**Please describe the purpose of this pull request.**
This is a partial fix for #755 for `local` embeddings. We might want to hold this until we can extend this fix to other embeddings endpoint types, but given that users are experiencing pain over this perhaps it is worth making it available now and fixing the others later.

The change checks length of the token sequence and whenever the sequence exceeds the model's limit it truncates the text to exactly the model limit.

**How to test**
It may be worthwhile to first test without this change so that you can locate instances that exhibit the #755 bug.
Try `memgpt load directory --recursive --name <some-name> --input-dir <your-dir>`
It's unclear to me which types of files are most problematic. I found it easiest to produce the problem with directories containing a lot of C++ code. But users have reported the problem with .pdf and I think .epub.

Once you have a directory that reproduces the bug, check out this branch and try again. However, you may also
need the new `--extensions <extension-list>` option. The default extensions currently configured in this branch is ".txt.,.md,.pdf". If you reproduced the bug with one of these file types then you don't need the option.

I run the test with `--extensions .md,.hpp,.cpp`

**Have you tested this PR?**
Yes. With the fix, the output should be similar to the output produced in the past when no files failed to load.

**Related issues or PRs**
#755 

**Is your PR over 500 lines of code?**
No

**Additional context**
Add any other context or screenshots about the PR here.
